salt/modules/network.py

```diff
@@ -1309,6 +1331,8 @@ def default_route(family=None):
     for route in _routes:
         if family:
             if route['destination'] in default_route[family]:
+                if __grains__['kernel'] == 'SunOS' and route['addr_family'] != family:
```

Comment:
_routes contain the output of netstat, the values depends on the platform.

As far as I know only SunOS variants have the addr_family field which is either inet or inet6.
In the case of SunOS de default route is identified by 'default' for both inet and inet6.

If family is set we need to filter on them, for SunOS we still get both inet and inet6 by just filtering on destination. So for SunOS we also check the addr_family property.